examples-standalone/personal-finance-app/src/app/components/settings/settings.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component } from '@angular/core';
+import { KENDO_INPUTS } from '@progress/kendo-angular-inputs';
+import { KENDO_LABELS } from '@progress/kendo-angular-label';
+import { KENDO_ICONS, SVGIcon } from '@progress/kendo-angular-icons';
+import { User } from '../../models/user';
+import { cardDetails, user } from '../../data/user-details';
+import { CardDetails } from '../../models/card-details';
+import { KENDO_DATEINPUTS } from '@progress/kendo-angular-dateinputs';
+import { cloudIcon } from '@progress/kendo-svg-icons';
+import { KENDO_DROPDOWNS } from '@progress/kendo-angular-dropdowns';
+import { KENDO_BUTTONS } from '@progress/kendo-angular-buttons';
+import {
+  FormBuilder,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { MessageService } from '@progress/kendo-angular-l10n';
+import { CustomMessagesService } from '../../services/custom-messages.service';
+
+@Component({
+  selector: 'app-settings',
+  standalone: true,
+  imports: [
+    ReactiveFormsModule,
+    KENDO_INPUTS,
+    KENDO_LABELS,
+    KENDO_DATEINPUTS,
+    KENDO_ICONS,
+    KENDO_DROPDOWNS,
+    KENDO_BUTTONS,
+  ],
+  templateUrl: './settings.component.html',
+  styleUrl: './settings.component.css',
+})
+export class SettingsComponent {
+  public user: User = user;
+  public cardDetails: CardDetails = cardDetails;
+  public userForm!: FormGroup;
+  public cardForm!: FormGroup;
+  public cloudIcon: SVGIcon = cloudIcon;
+  public countries: string[] = [];
+  private initialUserFormValues: any;
+  private initialCardFormValues: any;
+
+  public customMsgService: CustomMessagesService;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private messages: MessageService
+  ) {
+    this.customMsgService = this.messages as CustomMessagesService;
+    this.countries = Array.isArray(this.user.country)
+      ? this.user.country
+      : [this.user.country];
+    this.initUserForm();
+    this.initCardForm();
+
+    this.initialUserFormValues = { ...this.userForm.getRawValue() };
+    this.initialCardFormValues = { ...this.cardForm.getRawValue() };
```

Comment:
This should be tested.
```suggestion
    this.initialUserFormValues = { ...this.userForm.getRawValue() } as Partial<User>;
    this.initialCardFormValues = { ...this.cardForm.getRawValue() } as Partial<CardDetails>;
```